Google.Api.Generator.Tests/ProtoTest.cs

```diff
@@ -96,31 +96,66 @@ private void ProtoTestSingle(string testProtoName)
                 Assert.True(File.Exists(expectedFilePath), $"Expected file does not exist: '{expectedFilePath}'");
                 var expectedLines = File.ReadAllLines(expectedFilePath).Select(x => x.Trim('\r')).ToList();
                 var actualLines = Encoding.UTF8.GetString(file.Content).Split('\n').Select(x => x.Trim('\r')).ToList();
-                // Check that all the testable lines with in the expected file match.
-                int expectedIndex = 0;
-                int actualIndex = 0;
-                bool active = !expectedLines.Any(x => x.Trim() == "// TEST_START");
-                while (expectedIndex < expectedLines.Count && actualIndex < actualLines.Count)
+                var expectedBlocks = expectedLines.Any(x => x.Trim() == "// TEST_START") ? TestBlocks() : new[] { expectedLines.Select((s, i) => (i, s)).ToList() };
+                // Check that all expected code blocks in the expected code exist in the generated (actual) code.
+                // The order of the test blocks is not enforced, only the content.
+                foreach (var block in expectedBlocks)
                 {
-                    active &= expectedLines[expectedIndex].Trim() != "// TEST_END";
-                    if (active)
+                    int blockIndex = 0;
+                    (int lineNumber, string line) missing = default;
+                    int missingBestLength = -1;
```

Comment:
I don't really understand this code, to be honest. What's the "best length" here?